### PR TITLE
(WIP) Move unknown size calculation to FUSE

### PR DESCRIPTION
**Pull request by ekinanp**
This is where it's needed.

Fixes <TODO: Link issue #>

Signed-off-by: Enis Inan <enis.inan@puppet.com>

### CODE REVIEW

File: fuse/core.go [hunk 1]
```diff
@@ -162,6 +162,18 @@ func (f *fuseNode) Attr(ctx context.Context, a *fuse.Attr) error {
 		return err
 	}
 	attr := plugin.Attributes(updatedEntry)
+	if !attr.HasSize() && plugin.ReadAction().IsSupportedOn(updatedEntry) {
+		// This Readable entry doesn't know its size, so download its content and set the size
+		// to that. This enables built-in tools like `cat` to work without slowing down `list`.
+		//
+		// TODO: How can we prevent tab-completion from being slow?
```
Comment (ekinanp) [inline fuse/core.go]: Not sure how to do this. Could we check for tab-completion process by checking for the shell's PID?

Either way, I think moving this stuff over to FUSE is the right move. It seemed weird to include a size sometimes, and not include a size other times -- `plugin.Attributes` should return what the plugin author provided.

Comment (ekinanp) [reply fuse/core.go]: Also OK with deferring the fix for later since other users have not yet raised this issue.

Comment (MikaelSmith) [reply fuse/core.go]: Yeah, checking for the shell's PID might actually work.

Comment (ekinanp) [reply fuse/core.go]: Good news is this fix doesn't slow down `ls` or `find` since those use the API. Right now it's just tab-completion.

**Pull request closed without merging**